pkg/registry/node/etcd/etcd.go

```diff
@@ -102,3 +103,25 @@ var _ = rest.Redirector(&REST{})
 func (r *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
 	return node.ResourceLocation(r, r.connection, r.proxyTransport, ctx, id)
 }
+
+// An interface for types that allow getting information about a Node on which give pod is running.
+type HostLocator interface {
```

Comment:
I don't think we need this interface, the returned port is part of what ConnectionInfoGetter already returns... I'd rather have an implementation of that interface that takes `node.Status.DaemonEndpoints.KubeletEndpoint.Port` into account